x-pack/plugins/security_solution/server/lib/sourcerer/routes/helpers.ts

```diff
@@ -14,8 +14,14 @@ export const findExistingIndices = async (
   Promise.all(
     indices
       .map(async (index) => {
+        const splitIndex = index.split('');
+        const isLeadingDash = splitIndex.shift() === '-';
+        let indexToQuery = index;
+        if (isLeadingDash) {
+          indexToQuery = splitIndex.join('');
+        }
```

Comment:
Consider simplifying the following:

```ts
        const splitIndex = index.split('');
        const isLeadingDash = splitIndex.shift() === '-';
        let indexToQuery = index;
        if (isLeadingDash) {
          indexToQuery = splitIndex.join('');
        }
```

to eliminate mutations (and handle the case of leading whitespace) with something like:

```ts
        const indexToQuery = index.trim().startsWith('-')
          ? index.trim().substring(1)
          : index.trim();
```

To prove the above is equivalent, consider adding the following test, which also verifies leading whitespace is handled correctly:

```ts
  it('findExistingIndices removes leading / trailing whitespace, and dashes from exclude patterns', async () => {
    await findExistingIndices(
      [
        '   include-with-leading-and-trailing-whitespace    ',
        '   -exclude-with-leading-and-trailing-whitespace   ',
      ],
      esClient as unknown as ElasticsearchClient
    );
    expect(esClient.fieldCaps.mock.calls[0][0].index).toEqual(
      'include-with-leading-and-trailing-whitespace'
    );
    expect(esClient.fieldCaps.mock.calls[1][0].index).toEqual(
      'exclude-with-leading-and-trailing-whitespace'
    );
  });
```